src/redux/wallets/actions.ts

```diff
@@ -79,22 +79,20 @@ export const createHWWallet = (
 
         dispatch(verifyAddressOnDevice(true));
         const account = await wallet.getAccounts(blockchain, 0);
-        dispatch(
-            addWallet({
-                id: walletId,
-                hwOptions: {
-                    deviceId,
-                    deviceVendor,
-                    deviceModel,
-                    connectionType
-                },
-                name: `Wallet ${getState().wallets.length + 1}`,
-                type: WalletType.HW,
-                accounts: account.reduce((out, accounts) => {
-                    return out.concat(accounts);
-                }, [])
-            })
-        );
+        const walletData: IWalletState = {
+            id: walletId,
+            hwOptions: {
+                deviceId,
+                deviceVendor,
+                deviceModel,
+                connectionType
+            },
+            name: `Wallet ${Object.keys(getState().wallets).length + 1}`,
+            type: WalletType.HW,
+            accounts: account.reduce((out, accounts) => out.concat(accounts), [])
```

Comment:
it seems that the reduce concatenates accounts, in this case i think we can use accounts: account` or am I missing something ?

Reply:
![image](https://user-images.githubusercontent.com/32679726/71313656-6e973200-2444-11ea-8938-efcee9d3428c.png)

It seems that `accounts` it's an array of `IAccountState` arrays and the reduce pulls out into an array of objects.

Reply:
that's true, but i think you tried `createHDWallet` not `createHWWallet`.
In `createHWWallet` we have `const account = await wallet.getAccounts(blockchain, 0);`, in this case `account` should be an array of `IAccountState`.

Reply:
Yes indeed, I double checked and this is correct.`account` is basically `IAccountState[]`. I was looking at `createHDWallet`. I have updated to code. Sorry for the mistake.